tests/azure/nic_test.go

```diff
@@ -0,0 +1,69 @@
+package azure_test
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/NeowayLabs/klb/tests/lib/azure"
+	"github.com/NeowayLabs/klb/tests/lib/azure/fixture"
+)
+
+func genNicName() string {
+	return fmt.Sprintf("klb-nic-tests-%d", rand.Intn(1000))
+}
+
+func testNicCreate(t *testing.T, f fixture.F) {
+	vnet := genVnetName()
+	nsg := genNsgName()
+	nic := genNicName()
+	subnet := genSubnetName()
+	vnetAddress := "10.116.0.0/16"
+	addrnic := "10.116.1.100"
+	subnetAddress := "10.116.1.0/24"
+
+	f.Shell.Run(
+		"./testdata/create_nsg.sh",
+		nsg,
+		f.ResGroupName,
+		f.Location,
+	)
+	_ = azure.NewNsg(f)
+
+	f.Shell.Run(
+		"./testdata/create_vnet.sh",
+		vnet,
+		f.ResGroupName,
+		f.Location,
+		vnetAddress,
+	)
+	_ = azure.NewVnet(f)
+
+	f.Shell.Run(
+		"./testdata/create_subnet.sh",
+		subnet,
+		f.ResGroupName,
+		vnet,
+		subnetAddress,
+		nsg,
+	)
+	_ = azure.NewSubnet(f)
+
+	f.Shell.Run(
+		"./testdata/create_nic.sh",
+		f.ResGroupName,
+		nic,
+		f.Location,
+		vnet,
+		subnet,
+		addrnic,
+	)
+	nics := azure.NewNic(f)
+
+	nics.AssertExists(t, nic, nsg, addrnic)
```

Comment:
You configured the vnet/subnet/addrnic on the nic, it is enough to just validate the addnic ? Perhaps it is, since the addr is already on the subnet/vnet. Although, I think it is possible to have a different VNET with the same subnets/addrs, so a bug could slip in here (but it would be hard)